### DISCUSSION TITLE
How can sites detect a browser?

### DISCUSSION THREAD

**Discussion by zkisaboss** [denschub/chrome-mask#20] [Q&A] [answered]
How do websites like https://www.whatismybrowser.com/ know what my browser "looks like" even when changing my user agent

**Comment by skyschub** [comment, marked as answer]
This is a ... **very** complicated question.

There are a few more ways designed to identify a browser that this addon isn't spoofing. One of them is the [Client Hints](https://developer.mozilla.org/en-US/docs/Web/HTTP/Client_hints) headers, which would identify Chrome even with a UA spoof - and if Firefox sends a Chrome UA but no Client Hints, then it can't possibly be a real Chrome.

There's also a lot more advanced browser fingerprinting. You can, for example, check if the browser supports WebUSB. If it does not (like Firefox), but it sends a Chrome user agent, it clearly can't be Chrome. This is a very deep rabbit hole. :)
